ebpfsvc/rpc_api.cpp

```diff
@@ -51,3 +51,26 @@ ebpf_server_verify_and_load_program(
     return EBPF_OPERATION_NOT_SUPPORTED;
 #endif
 }
+
+ebpf_result_t
+ebpf_server_verify_and_authorize_native_image(
+    /* [string][in] */ char* image_path)
+{
+    HANDLE native_image_handle = INVALID_HANDLE_VALUE;
+    ebpf_result_t result;
+    if (RpcImpersonateClient(nullptr) != RPC_S_OK) {
+        return EBPF_ACCESS_DENIED;
+    }
+    result = ebpf_verify_signature_and_open_file(image_path, &native_image_handle);
```

Comment:
why not encapsulate the rpc impersonation, revert and auth_native_module calls inside this function?